### DISCUSSION TITLE
Rectified pentachoron

### DISCUSSION THREAD

**Discussion by mecejide** [SupposeNot/RAMP#184] [Q&A] [answered]
How do I construct a rectified pentachoron in RAMP?

**Comment by SupposeNot** [comment]
Currently we don't have general truncation operations implemented in higher ranks. This leaves you essentially two choices I can think of (short of hand coding the connection group... don't recommend that one).

One option is to use WythoffVoltageOperator. The documentation about how this works is still kind of thin, and Mark is probably the one to give you more insight into that. This would necessitate figuring out the voltage graph and lifting group for this example.

A second option is to use PosetFromAtomicList (where you would list the vertices in a master list of facets) in combination with ConnectionGroup and AbstractPolytope. Here's an example for the pyramid over the square:
gap> facets:=[[1,2,3,4],[1,2,5],[2,3,5],[3,4,5],[1,4,5]];
[ [ 1, 2, 3, 4 ], [ 1, 2, 5 ], [ 2, 3, 5 ], [ 3, 4, 5 ], [ 1, 4, 5 ] ]
gap> PosetFromAtomicList(facets);
A poset.
gap> p:=AbstractPolytope(ConnectionGroup(last));
3-polytope with 32 flags
gap> SchlafliSymbol(p);
[ [ 3, 4 ], [ 3, 4 ] ]
gap> p=Pyramid(4);
true

**Comment by CunningGabe** [comment, marked as answer]
I think this does what you want:
M := WythoffVoltageOperator(4, [1], Simplex(4));

In general, WythoffVoltageOperator(rank, ListOfRingedNodes, BaseManiplex) performs the appropriate Wythoff operation. It does the voltage stuff behind the scenes.

(We should probably just rename this to Wythoffian...)

**Reply by SupposeNot** [reply]
I think we need to expand the documentation a bit so this kind of thing is clearer, i.e., what does the 4 and the [1] do? This is a place in the documentation where a bunch of worked examples (and possibly a diagram or two) would be helpful.

**Reply by CunningGabe** [reply]
Indeed - I did a bit of cleanup of this this morning now, so I hope it's clearer.

I'll add an issue to add some convenience methods like Rectification, Truncation (for general rank), etc.
